client-course-schedulizer/src/components/App/App.tsx

```diff
@@ -13,6 +21,68 @@ export const App = () => {
   const [isCSVLoading, setIsCSVLoading] = useState(false);
   const { save } = useLocal("appState");
   const [appState, appDispatch] = useReducer(reducer(save), initialAppState);
+  const { schedule, fileUrl } = appState;
+
+  // Load a file from the GET parameters (either ?csv= or ?xlsx=),
+  // only do this once for each page refresh and only do this if the initial state is empty
+  // Ref: https://stackoverflow.com/questions/56663785/invalid-hook-call-hooks-can-only-be-called-inside-of-the-body-of-a-function-com
+  // Ref: https://www.gitmemory.com/issue/SheetJS/js-xlsx/1110/531003740
+  useEffect(() => {
```

Comment:
This effect is good, but I would move this effect into its own hook, like `useScheduleFromUrl()` and then call that in `App.tsx`. This separates the logic of getting the schedule from the app setup.

Reply:
686213d but I called it in `SchedulizerPage.tsx`